geomstats/information_geometry/poisson.py

```diff
@@ -0,0 +1,368 @@
+        bound : float
+            Right-end ot the segment where Poisson parameters are sampled.
+            Optional, default: 1.
+
+        Returns
+        -------
+        samples : array-like, shape=[n_samples,]
+            Sample of points representing Poisson distributions.
+        """
+        size = (n_samples, self.dim) if n_samples != 1 else (self.dim,)
+        return bound * gs.random.rand(*size)
+
+    def projection(self, point, atol=gs.atol):
+        """Project a point in ambient space to the open set.
+
+        The last coordinate is floored to `gs.atol` if it is non-positive.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point in ambient space.
+        atol : float
+            Tolerance to evaluate positivity.
+
+        Returns
+        -------
+        projected : array-like, shape=[..., 1]
+            Projected point.
+        """
+        return gs.where(point < atol, atol, point)
+
+    def sample(self, point, n_samples=1):
+        """Sample from the Poisson distribution.
+
+        Sample from the Poisson distribution with parameter provided
+        by point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point representing an Poisson distribution.
+        n_samples : int
+            Number of points to sample with each parameter in point.
+            Optional, default: 1.
+
+        Returns
+        -------
+        samples : array-like, shape=[..., n_samples]
+            Sample from Poisson distributions.
+        """
+
+        def _sample(param):
+            return poisson.rvs(param, size=n_samples)
+
+        if point.ndim > 1:
+            return gs.array([_sample(point_) for point_ in point])
+
+        return gs.array(_sample(point))
+
+    def point_to_pdf(self, point):
+        """Compute pmf associated to point.
+
+        Compute the probability mass function of the Poisson
+        distribution with parameters provided by point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point representing an Poisson distribution (scale).
+
+        Returns
+        -------
+        pdf : function
+            Probability mass function of the Poisson distribution with
+            scale parameter provided by point.
+        """
+
+        def pmf(k):
+            """Generate parameterized function for Poisson pmf.
+
+            Compute the probability mass function of the Poisson
+            distribution with parameters provided by point.
+
+            Parameters
+            ----------
+            k : array-like, shape=[n_points,]
+                Point representing an Poisson distribution (lambda).
+
+            Returns
+            -------
+            pmf_at_k : array-like, shape=[..., n_points]
+                Probability mass function of the Poisson distribution with
+                parameters provided by point.
+            """
```

Comment:
Same comments as for the geometric distributions: samples and not points representing Poisson distributions.